src/ic-cdk/src/api/management_canister/http_request.rs

```diff
@@ -200,6 +243,47 @@ pub async fn http_request_with_cycles(
     .await
 }
 
+/// Make an HTTP request to a given URL and return the HTTP response, after a transformation.
+///
+/// Do not set the `transform` field of `arg`. To use a Candid function, call [`http_request_with_cycles`] instead.
+///
+/// See [IC method `http_request`](https://internetcomputer.org/docs/current/references/ic-interface-spec/#ic-http_request).
+///
+/// This call requires cycles payment. The required cycles is a function of the request size and max_response_bytes.
+/// Check [this page](https://internetcomputer.org/docs/current/developer-docs/production/computation-and-storage-costs) for more details.
+///
+/// If the canister is on a 13-node Application Subnet, you can call [`http_request`] instead which handles cycles cost calculation under the hood.
+#[cfg(any(docsrs, feature = "transform-closure"))]
+#[cfg_attr(docsrs, doc(cfg(feature = "transform-closure")))]
+pub async fn http_request_with_cycles_with(
+    arg: CanisterHttpRequestArgument,
+    cycles: u128,
+    transform_func: impl FnOnce(HttpResponse) -> HttpResponse + 'static,
+) -> CallResult<(HttpResponse,)> {
+    assert!(arg.transform.is_none());
```

Comment:
The `assert` here might cause `trap` without giving useful information.
It seems that there is no way to guarantee that the `arg.transform.is_none()` at compile time.
I would prefer to remove this `assert`.
We clearly state this behavior in the doc comment above.

Reply:
If a user specifies a transform function, it will be replaced without being called. Better to fail loudly than to fail silently imo.

Reply:
In other words, a trap caused by incorrectly written code will never be useful to the canister's caller - the most we can hope for is to be useful to the canister's implementor, which a panic would be.

Reply:
What will the error look like? Can the implementor easily figure out the reason of trap?

Reply:
It would print `arg.transform.is_none()`. Maybe I should add a more descriptive message since `arg` is not exactly eye-catching even though it's in the documentation.